distribution/kpkginstall/runtest.sh

```diff
@@ -5,34 +5,25 @@
 ARCH=$(uname -m)
 REBOOTCOUNT=${REBOOTCOUNT:-0}
 
-# Output version of the kernel contained in the specified kernel package
-# tarball.
-#
-# Args: tarball
-# Output: kernel version, nothing if not found
-# Status: zero if version is found, non-zero on failure
 function get_kpkg_ver()
 {
-  case "$KPKG" in
+  echo "Extracting kernel version from ${KPKG_URL}" | tee -a ${OUTPUTFILE}
+
+  case "${KPKG_URL}" in
     *.tar.gz)
-            tar tf "$1" | sed -ne '/^boot\/vmlinu[xz]-[1-9]/ {s/^[^-]*-//p;q}; $Q1'
+            KPKG=${KPKG_URL##*/}
+            tar tf "$KPKG" | sed -ne '/^boot\/vmlinu[xz]-[1-9]/ {s/^[^-]*-//p;q}; $Q1'
             ;;
     *)
-            # replace '-' with ' ' and use natural IFS to split into an array
-            karray=(${1//-/ })
-            if [ "${#karray[@]}" == "3" ]; then
-              echo "${karray[1]}-${karray[2]}.$ARCH"
-            elif [ "${#karray[@]}" == "4" ]; then
-              # uname -r have variants at the end
-              KVAR="${karray[1]}"
-              echo "${karray[2]}-${karray[3]}$KVAR.$ARCH"
-            fi
+            # Grab the kernel version from the provided repo directly
+            dnf --disablerepo="*" --enablerepo="kernel-cki" list kernel | grep kernel-cki | awk '{print $2}'
```

Comment:
You could also write `awk '/kernel-cki/ {print $2}'` instead of `grep kernel-cki | awk '{print $2}'`.